bin/grommet.js

```diff
@@ -17,76 +18,159 @@ String.prototype.capitalize = function() {
-        var shortcutIcon = path.join(grommetPath, 'img/shortcut-icon.png');
-
-        gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
-        gulp.src(templateFolder, { dot: true })
-        .pipe(template({
-          appName: app,
-          appTitle: title,
-          grommetVersion: packageJSON.version
-        }))
-        .pipe(gulp.dest('./'))
-        .pipe(install()).on('finish', function() {
-          done();
-        });
+      if (fileExists(newAppPath)) {
+        console.error('[grommet] Error while creating new app. Directory "'+command.options.app+'" already exists.');
+        process.exit(1);
       }
-    });
+
+      mkdirp(newAppPath, function(err) {
+        if (err) {
+          console.error('[grommet] Error trying to create project: ' + err);
+          process.exit(1);
+        }
+
+        try {
+          var templateFolder = path.join(command.env.templatePath, 'new/**');
+          copyAssetsAndInstall(templateFolder, newAppPath, command);
+        } catch(err) {
+          shelljs.rm('-rf', newAppPath);
+          console.error('[grommet] Error while creating new app '+command.options.app+'.');
+          throw err;
+        }
+      })
+    }
   }
 };
 
+var supportedNodeVersion = '0.10';
+var supportedNpmVersion = '1.4';
 var grommetPath = path.join(__dirname, '..');
-var packageJSON = require(path.join(grommetPath, 'package.json'));
-var npmVersion = Number(shelljs.exec('npm --version', {silent:true})
-  .stdout.toString().match(/^(\d+\.\d+)/)[1]);
 var argv = require('yargs').boolean('version').argv;
+var options = argv._;
+var command = {
+  task: options[0] || 'init',
+  env: {
+    grommetPath: grommetPath,
+    grommetVersion: require(path.join(grommetPath, 'package.json')).version,
+    templatePath: path.join(grommetPath, 'templates'),
+    npmVersion: Number(shelljs.exec('npm --version', {silent:true}).stdout.toString().match(/^(\d+\.\d+)/)[1]),
+    nodeVersion: Number(process.version.match(/^v(\d+\.\d+)/)[1]),
+    processVersion: process.version
+  },
+  options: {}
+}
 
 if (argv.version) {
-  console.log(packageJSON.version);
+  console.log(command.env.grommetVersion);
   process.exit(0);
 }
 
-var options = argv._;
-var command = options[0] || 'init';
-if (command === 'init') {
-  var app = options[1] || 'app-name';
-  var title = app.replace(/-|_/g, ' ').capitalize();
+switch(command.task) {
+  case 'init':
+    command.options['app'] = options[1] || 'app-name';
+    command.options['title'] = command.options.app.replace(/-|_/g, ' ').capitalize();
+    break;
+  case 'new':
+    command.options['app'] = options[1];
+    if (!command.options.app) {
+      console.error('[grommet] Usage: grommet new <app-name> [description]');
+      process.exit(1);
+    }
+    command.options['title'] = command.options.app.replace(/-|_/g, ' ').capitalize();
+    command.options['description'] = options[2] || "";
```

Comment:
> how you deal with spaces in the description ?

If the user wants to input a description that contains white spaces, he should surround the description with quotes, since the description is a command line option,